### ISSUE TITLE
Should we change book page language selector?

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#8751]
url: https://openlibrary.org/works/OL82560W/Harry_Potter_and_the_Goblet_of_Fire

Currently, we show which other editions exist with previews in other languages, near the metadata "omnibar"
<img width="484" alt="Screenshot 2024-01-24 at 9 07 43 AM" src="https://github.com/internetarchive/openlibrary/assets/978325/b27f534b-842c-4e24-964c-6e39b1e0847c">
This makes sense because the omnibar shows the current edition's language

However (a) it may be more appropriate near the editions table, where one may change which edition is selected and (b) the format/UI may be more useful as e.g. a dropdown menu/selector.

<img width="746" alt="Screenshot 2024-01-24 at 9 10 35 AM" src="https://github.com/internetarchive/openlibrary/assets/978325/570e4ee7-6ea9-4a52-9266-c4345804b4d9">

This issue asks for designs from the community, and discussion on both the UI and the position of this language preview data.

**Comment by vamshipy** [comment]
hi, do you want to have a languages tab beside the show entries tab and remove the "preview in other languages" feature.. would love to work on this

**Comment by jimchamp** [comment]
@vamshipy we don't know.  This is a call for designs.  Feel free to link to a Figma mock-up with your proposal.

**Comment by vamshipy** [comment]
[https://www.figma.com/proto/IWC7lt00agCGaYSSzhzg2T/Untitled?type=design&node-id=1-2&t=OxMcSnuNCxvgnIFF-1&scaling=min-zoom&page-id=0%3A1&mode=design](url) ..is this what you want it to look like

**Comment by mekarpeles** [comment]
I think we're agreed that language selector should move into the Editions table area, as opposed to below the omnibar.

Near the editions table, maybe we show 1st 3 languages in the list (arbitrarily) and then a dropdown for the rest: 
`Languages: [English] | [German] | [French] | [+7 others v]`

Bonus: If the "[+7 others v]" dropdown is searchable

We could prioritize based on the patron's browser language, if we can access it.

**Comment by Moksh45** [comment]
Hi, I would like to work on this issue

**Comment by jimchamp** [comment]
I've added this to the agenda of our next community design call, on the 16th.  We'll gather feedback on these designs during that meeting, and update this thread with the decisions.

**Comment by Moksh45** [comment]
okay

**Comment by DebbieSan** [comment]
Hi @Moksh45 @jimchamp, since we didn't have enough time to discuss this issue within our call this week, I am giving my feedback here.

The language selector on a dropdown menu makes sense as proposed by Moksh45. 

However, in regards to the search functionality suggested, are we going to add the functionality to search for the appropriate language within the dropdown menu? The search box to the right already searches for editions in another language. For example, if you type " French", it will find an edition in French, if I am correct. Would having another search in this area be a) redundant to the point of distracting, b)confusing or c)redundant to the point of useful to patrons as they will have multiple areas to search for a language?

Otherwise, this dropdown seems simple yet easier to use than the current design, wtg!

Cheers :)

**Comment by Realmbird** [comment]
Not sure I see people working on it is this still beginner friendly
If no one is assigned can this be assigned to me since this seems to be a UI problem I think I might be able to do it

**Comment by Realmbird** [comment]
If possible "assign me"

**Comment by Moksh45** [comment]
Hi @DebbieSan, thank you for sharing your feedback. It's great to see the positive response to the proposed language selector on the dropdown menu. Regarding the search functionality suggestion, it's important to consider user experience and efficiency.

Adding a search functionality specifically for languages within the dropdown menu could potentially enhance user convenience, especially if there are a large number of language options available. However, it's crucial to assess whether this additional feature would indeed add value or if it might lead to redundancy or confusion.

To determine the best approach, we could consider conducting user testing or gathering feedback from a sample group to understand their preferences and usability expectations. This way, we can ensure that any enhancements made align with user needs and preferences while maintaining simplicity and ease of use.

Thank you for raising this point

Cheers! 😄

**Comment by Realmbird** [comment]
@jimchamp could I work on this?

**Comment by Realmbird** [comment]
If I understand the problem is this what we are supposed to make
We are trying to make dropdown on side of editions so they can query the language of the book?
![Screenshot from 2024-03-08 10-44-44](https://github.com/internetarchive/openlibrary/assets/56459277/00d90338-b2a7-4a85-b157-e284359a41b4)
 
![Screenshot from 2024-03-08 10-31-02](https://github.com/internetarchive/openlibrary/assets/56459277/b2380837-57e4-44fa-a501-838bc3591ddf)
![Screenshot from 2024-03-08 10-32-17](https://github.com/internetarchive/openlibrary/assets/56459277/c0b2d3fd-b40d-4fd0-b262-b34597327894)

**Comment by mekarpeles** [comment]
@Moksh45 if you're interested, I think your solution makes sense to run with for now!

**Comment by mekarpeles** [comment]
After discussion during 06-07 design call, we converged on a design that looks broadly like:

`Show [10 | v] results per page`
`Show [ All | ebooks only | v ]`
`Language [ All | Albania | Chinese | English | ... | v]`

The next steps should be a PR that tests out this concept as a prototype given our current system.

Eventually we want to switch the editions table to use solr so we can get all languages, not just those that appear from e.g. the first 10 ebooks, see #7451

**Comment by aarasawa** [comment]
Can I work on this?

**Comment by jimchamp** [comment]
Thanks for your patience, @aarasawa.  Do you still want to work on this?

**Comment by aarasawa** [comment]
@jimchamp yes, I would

**Comment by jimchamp** [comment]
@aarasawa, you've been assigned.  The root editions table template can be found [here](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/type/work/editions_datatable.html).  The existing language preview list can be found [here](https://github.com/internetarchive/openlibrary/blob/79dccb33ad74f3e9f16e69dce2bae7a568c8d3d0/openlibrary/templates/type/edition/view.html#L323-L331).

**Comment by PredictiveManish** [comment]
Hey does anyone working on this issue or else I can start with this?

**Comment by jimchamp** [comment]
Thanks @PredictiveManish!

Before making code changes, can you help us with breakdown by:
1. Clarifying what you believe the task to be
2. Identifying which files are related to the issue
3. Asking any questions you may have about the goal or requirements of this issue
4. Proposing a solution or approach

**Comment by Anmol-77** [comment]
can i work on this?

**Comment by Sneha-a10** [comment]
Hi, if no one is working on this issue, I'd like to take it and start on it.

**Comment by jimchamp** [comment]
Anybody can provide a design for a different language selection flow for book page editions tables.  No specific person will be assigned to this issue -- just include your thoughts/designs in this thread.

**Comment by hornc** [comment]
Only slightly tongue-in-cheek; can we close this due to [Betteridge's law of headlines](https://en.wikipedia.org/wiki/Betteridge%27s_law_of_headlines)?   

> "Any headline that ends in a question mark can be answered by the word no."

It is very unclear what problem is being solved here (the non-technical outcome desired), who it benefits, or how it helps OL's mission.

I had to read the description multiple times to understand which page element is being talked about, and still do not know how we'd evaluate if a particular change was successful. 

These sorts of open ended issues generate a lot more discussion than other potentially more valuable and actionable feature requests, but it's unclear how they can be usefully resolved.

It looks like we already have a feature to filter only _previewable_ books by language. This appears to work and helps patrons locate readable books.  It doesn't filter the full list though, which does seem inconsistent as other filters target one facet at a time. Bringing the focus back to WHO needs to accomplish a GOAL, and WHY it's important will help with problem solving, if there is a problem to be solved.

Driving language filtering changes by #9128,  #9017, #8971 would be more productive, but even some of these could benefit from clearer focus on the outcome and value delivered. These are all old issues which haven't had much movement. I think focusing on service objectives, perhaps framed as agile user stories, would reduce the issue backlog by making it clear how requests are actionable.

**Comment by momna-ahmad** [comment]
Hey @mekarpeles @jimchamp I made some designs according to mentioned requirements in comments and would appreciate feedback 

<br>
<img width="1280" height="873" alt="Image" src="https://github.com/user-attachments/assets/1b669812-c169-4ee6-aecf-9753dd8a56e1" />
</br>

I have also included the search bar in dropdown as @mekarpeles proposed for searching available languages of editions

<br>
<img width="1280" height="873" alt="Image" src="https://github.com/user-attachments/assets/d16168d1-d02f-4b1a-8f5d-564ed40121dc" />
</br>

Let me know if the feature is accepted for implementation and any guidance for the implementation. If approved I can start working on it right away

**Comment by jimchamp** [comment]
I fully agree with you @hornc -- I've let this issue fall to the wayside due to not understanding how to proceed or what outcome we're aiming for.  Additionally, there's been some talk of replacing the edition able with a completely different experience.

I'm going to close this for now.  If we do end up keeping the edition table, we can circle back to the designs that were submitted here.
